### PR TITLE
[Feature][Mongodb-CDC] Support multi-table read

### LINKED ISSUE #7951: [Feature][CDC] Support multi table read for mongodb source
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Currently mongo cdc does not support multi-table reads. This is an interesting feature. Who is interested in completing it?

https://github.com/apache/seatunnel/tree/dev/seatunnel-connectors-v2/connector-cdc

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by hawk9821**
### Purpose of this pull request
support multiTable Mongodb-CDC

close #7951

### How was this patch tested?
E2E: MongodbCDCIT#testMongodbCdcMultiTableToMysqlCheckDataE2e

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by hawk9821** [comment]
image

File: seatunnel-examples/seatunnel-engine-examples/pom.xml [hunk 1]
```diff
@@ -67,5 +67,11 @@
             <version>${project.version}</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>connector-mongodb</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+
```
Comment (Carl-Zhou-CN) [inline seatunnel-examples/seatunnel-engine-examples/pom.xml]: delete

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mongodb-e2e/src/test/java/mongodb/MongodbCDCIT.java [hunk 2]
```diff
@@ -295,12 +312,35 @@ private List<List<Object>> querySql() {
...
     }
 
+    private void truncateMysqlTable(String tableName) {
+        String checkTableExistsSql =
+                "SELECT COUNT(*) FROM information_schema.tables WHERE table_schema = ? AND table_name = ?";
+        String truncateTableSql = String.format("TRUNCATE TABLE %s", tableName);
+
+        try (Connection connection = getJdbcConnection();
+                PreparedStatement checkStmt = connection.prepareStatement(checkTableExistsSql)) {
+            checkStmt.setString(1, MYSQL_DATABASE); // 或者你可以直接提供数据库名
```
Comment (Carl-Zhou-CN) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mongodb-e2e/src/test/java/mongodb/MongodbCDCIT.java]: English support

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/schema/TableSchemaOptions.java [hunk 3]
```diff
@@ -55,6 +55,12 @@ public static class TableIdentifierOptions {
                     .noDefaultValue()
                     .withDescription("SeaTunnel Schema");
 
+    public static final Option<List<Map<String, Object>>> TABLES_CONFIGS =
```
Comment (Carl-Zhou-CN) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/schema/TableSchemaOptions.java]: tables_configs has also appeared elsewhere

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/CatalogTableUtil.java [hunk 4]
```diff
@@ -111,6 +111,16 @@ public static List<CatalogTable> getCatalogTables(
...
+        List<Map<String, Object>> schemaMaps =
+                readonlyConfig.get(TableSchemaOptions.TABLES_CONFIGS);
+        if (schemaMaps != null) {
+            if (schemaMaps.isEmpty()) {
+                throw new SeaTunnelException("tables_configs can not be empty");
+            }
+            return schemaMaps.stream()
+                    .map(map -> buildWithConfig(factoryId, ReadonlyConfig.fromMap(map)))
+                    .collect(Collectors.toList());
+        }
```
Comment (Carl-Zhou-CN) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/CatalogTableUtil.java]: This is a method marked as @Deprecated. Can it be moved elsewhere?

File: seatunnel-examples/seatunnel-engine-examples/src/main/java/org/apache/seatunnel/example/engine/SeaTunnelEngineLocalExample.java [hunk 5]
```diff
@@ -36,7 +36,7 @@ public class SeaTunnelEngineLocalExample {
 
     public static void main(String[] args)
             throws FileNotFoundException, URISyntaxException, CommandException {
-        String configurePath = args.length > 0 ? args[0] : "/examples/fake_to_console.conf";
+        String configurePath = args.length > 0 ? args[0] : "/examples/mongodb_source.conf";
```
Comment (Carl-Zhou-CN) [inline seatunnel-examples/seatunnel-engine-examples/src/main/java/org/apache/seatunnel/example/engine/SeaTunnelEngineLocalExample.java]: recover

File: seatunnel-examples/seatunnel-engine-examples/src/main/resources/log4j2.properties [hunk 6]
```diff
@@ -19,7 +19,7 @@
 # The minimum amount of time, in seconds, that must elapse before the file configuration is checked for changes.
 monitorInterval = 60
 
-rootLogger.level = DEBUG
+rootLogger.level = INFO
```
Comment (Carl-Zhou-CN) [inline seatunnel-examples/seatunnel-engine-examples/src/main/resources/log4j2.properties]: recover

File: seatunnel-examples/seatunnel-engine-examples/pom.xml (hunk 1 above)
Comment (hawk9821) [reply seatunnel-examples/seatunnel-engine-examples/pom.xml]: get

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/CatalogTableUtil.java (hunk 4 above)
Comment (hawk9821) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/CatalogTableUtil.java]: get , the parsing logic is moved to `MongodbIncrementalSourceFactory`

File: seatunnel-examples/seatunnel-engine-examples/src/main/java/org/apache/seatunnel/example/engine/SeaTunnelEngineLocalExample.java (hunk 5 above)
Comment (hawk9821) [reply seatunnel-examples/seatunnel-engine-examples/src/main/java/org/apache/seatunnel/example/engine/SeaTunnelEngineLocalExample.java]: test code forgot to delete

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/schema/TableSchemaOptions.java (hunk 3 above)
Comment (hawk9821) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/schema/TableSchemaOptions.java]: In `FakeOption` and `BaseSourceConfigOptions` , it is used for `connector-fake` and `connector-file` respectively, so this configuration is added to TableSchemaOptions , whether unified configuration is necessary ?

Comment (Carl-Zhou-CN) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/schema/TableSchemaOptions.java]: @hailin0 cc

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mongodb-e2e/src/test/java/mongodb/MongodbCDCIT.java (hunk 2 above)
Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mongodb-e2e/src/test/java/mongodb/MongodbCDCIT.java]: +1

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/schema/TableSchemaOptions.java (hunk 3 above)
Comment (hawk9821) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/schema/TableSchemaOptions.java]: Unified tables_configs and table_list  #8100

File: docs/en/connector-v2/source/MongoDB-CDC.md [hunk 7]
```diff
@@ -213,6 +217,67 @@ sink {
...
+        }
+      }
+    ]
+  }
+}
+
+# Console printing of the read Mongodb data
+sink {
+  Console {
+    parallelism = 1
```
Comment (hailin0) [inline docs/en/connector-v2/source/MongoDB-CDC.md]:
```suggestion
```

**Review by Carl-Zhou-CN — DISMISSED** [review]
LGTM

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mongodb-e2e/src/test/resources/mongodbcdc_to_mysql.conf [hunk 8]
```diff
@@ -29,6 +29,11 @@ source {
     username = superuser
     password = superpw
     schema = {
+      table = "inventory.products"
+      primaryKey {
+        name = "id"
+        columnNames = ["_id"]
+      }
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mongodb-e2e/src/test/resources/mongodbcdc_to_mysql.conf]: Can the old config without table execute normally?

Comment (hawk9821) [reply seatunnel-e2e/seatunnel-connector-v2-e2e/connector-cdc-mongodb-e2e/src/test/resources/mongodbcdc_to_mysql.conf]: Modified to be compatible with older versions of single table synchronization

**Review by Hisoka-X — APPROVED** [review]

**Comment by Hisoka-X** [comment]
Thanks @hawk9821

**Review by Carl-Zhou-CN — APPROVED** [review]
+1

File: seatunnel-connectors-v2/connector-cdc/connector-cdc-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mongodb/source/offset/ChangeStreamOffset.java
Comment (jw-itq) [inline seatunnel-connectors-v2/connector-cdc/connector-cdc-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/cdc/mongodb/source/offset/ChangeStreamOffset.java:1]: may I ask if this modification can solve the issue(https://github.com/apache/seatunnel/issues/7520)

**Pull request merged**